File: 0301-0400/0374-Guess Number Higher or Lower/0374-Guess Number Higher or Lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        low = 1
        high = n
        while low <= high:
            mid = (low + high) // 2
            feedback = guess(mid)
            if feedback == -1:
                high = mid - 1
            elif feedback == 1:
                low = mid + 1
            else:
                return mid

        return low

File: 0301-0400/0374-Guess Number Higher or Lower/0374-Guess Number Higher or Lower.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        int low = 1, high = n;
        while (low <= high) {
            int mid = low + (high - low) / 2;
            int feedback = guess(mid);
            if (feedback == -1) {
                high = mid - 1;
            }
            else if (feedback == 1) {
                low = mid + 1;
            }
            else {
                return mid;
            }
        }
        
        return low;
    }
};
